arkouda_integration/client/arkouda_integration/k8s.py

```diff
@@ -0,0 +1,294 @@
+
+from kubernetes import client  # type: ignore
+from kubernetes.client import (  # type: ignore
+    V1ObjectMeta,
+    V1Pod,
+    V1Service,
+    V1ServicePort,
+    V1ServiceSpec,
+)
+from kubernetes.client.rest import ApiException  # type: ignore
+from arkouda.logger import getArkoudaLogger
+
+logger = getArkoudaLogger(name="Arkouda Integration Client")
+
+class InvocationMethod(Enum):
+    """
+    The InvocationMethod enum provides controlled vocabulary
+    for KubernetesDao methods called from scripts
+    """
+    GET_POD_IPS = 'GET_POD_IPS'
+    GET_PODS = 'GET_PODS'
+
+    def __str__(self) -> str:
+        """
+        Overridden method returns value, which is useful in outputting
+        an object to JSON.
+        """
+        return self.value
+
+    def __repr__(self) -> str:
+        """
+        Overridden method returns value, which is useful in outputting
+        an object to JSON.
+        """
+        return self.value
+
+
+class DaoError(Exception):
+    """
+    The DaoError class wraps tech-specific errors
+    """
+    pass
+
+
+class KubernetesDao:
+    """
+    The KubernetesDao class encapsulates metadata and methods used to retrieve
+    objects such as pods and services from Kubernetes.
+    """
+    __slots__ = ("core_client", "apps_client")
+
+    core_client: client.CoreV1Api
+    apps_client: client.AppsV1Api
+
+    def __init__(self, cert_file : str, key_file : str, k8s_host : str):
+        config = client.Configuration()
+
+        config.cert_file = cert_file
+        config.key_file = key_file
+        config.host = k8s_host
+
+        # used to disable non-verified cert warnings
+        config.verify_ssl = False
+
+        import urllib3  # type: ignore
+
+        urllib3.disable_warnings()
+
+        try:
+            api_client = client.ApiClient(configuration=config)
+            self.core_client = client.CoreV1Api(api_client=api_client)
+            self.apps_client = client.AppsV1Api(api_client)
+        except Exception as e:
+            raise DaoError(e)
+
+    def get_pods(self, namespace: str, app_name: Optional[str] = None) -> List[V1Pod]:
+        """
+        Retrieves a list of V1Pod objects corresponding to pods within a
+        namespace. An app name is optionally provided to narrow the scope
+        of pods returned.
+
+        :param str namespace: namespace to be queried
+        :param Optional[str] app_name: name of app corresponding to the pods
+        :return: a list of pods
+        :rtype: List[V1Pod]
+        :raises: DaoError if there is an error in retrieving pods
+        """
+        try:
+            if namespace:
+                pods = self.core_client.list_namespaced_pod(namespace=namespace)
+                if app_name:
+                    filteredPods = []
+                    for pod in pods.items:
+                        if self._is_named_pod(pod, app_name):
+                            filteredPods.append(pod)
+                    return filteredPods
```

Comment:
this is just stylistic but this can be done with a list expression just like the other returns in this function
```python
return [pod for pod in pods.items if self._is_named_pod(pod, app_name)]
```

Reply:
@pierce314159 agreed, will fix

Reply:
@pierce314159 done! thanks for catching this.